deploy/crds/infinispan.org_infinispans_crd.yaml

```diff
@@ -656,6 +656,7 @@ spec:
                   - NodePort
                   - LoadBalancer
                   - Route
+                  - None
```

Comment:
@dmvolod Would "Internal" be more meaningful than "None"?

Reply:
Yes, correct @oraNod I was looking for more relevant meaning.
Probably, we need to split expose types for xsite and cluster services, as some values are not allowed for xiste, some for service and these setup could confuse customers.

Reply:
Isn't there already a default "internal" service that the Operator creates? Could that be used to achieve the same functionality that this PR offers?

Reply:
Yes and no, the ports and names of these services are different. We could try to utilize them but this adds additional complexity.
Another option is skip expose type for internal service. In this `case spec.expose.type` need to make optional.

Reply:
Yeah. Maybe I need more time for my brain to kick in after PTO but either way it seems like the changes in this PR introduce complexity. Is the functionality gained worth it? Seems like we already have the manual xsite config. Could you add the internal IP of the cluster and get the same?

Reply:
> Yeah. Maybe I need more time for my brain to kick in after PTO but either way it seems like the changes in this PR introduce complexity. Is the functionality gained worth it? Seems like we already have the manual xsite config. Could you add the internal IP of the cluster and get the same?

Sometimes yes, sometimes no. For example on AWS or Azure you don't know about node IP's if you don't control whole env explicitly. This gives much more flexibility to customers.

Reply:
I would avoid "None" since it seems that nothing is exposed. 
I probably choose "ClusterIP" too keep it consistent with openshift names, because we already have:
```
ispnv1.ExposeTypeNodePort => corev1.ServiceTypeNodePort
ispnv1.ExposeTypeLoadBalancer => corev1.ServiceTypeLoadBalancer
```

Reply:
> I probably choose "ClusterIP" too keep it consistent with openshift names, because we already have:

Good, make sense. I will try to think about splitting expose.type for Cross-Site and REST/HR endpoints, as Router doesn't make sense for Cross-Site, but ClusterIP service doesn't make sense for REST/HR and always created.

Reply:
`ClusterIP` for REST/HR could make sense if the application is running in the same namespace, like EAP session externalization with RHDG :) 
I agree to keep `ClusterIP` away from REST/HR

Reply:
Yeah, agree.
Another reason of splitting that customers confusing Route option for cross-site which is not possible